src/Lorentz/Contracts/BaseDAO/Doc.hs

```diff
@@ -0,0 +1,168 @@
+  |]
+
+acceptOwnershipDoc :: Markdown
+acceptOwnershipDoc = [md|
+  Accepts the administrator privelege.
+  Only works when sender was asked to become an admin and only if it was asked by current admin.
+  |]
+
+migrateDoc :: Markdown
+migrateDoc = [md|
+  Asks an address to migrate the contract to it.
+  Can be performed only by current administrator.
+  The contract is not considered migrated, until it recieves confirm_migration call.
+  |]
+
+confirmMigrationDoc :: Markdown
+confirmMigrationDoc = [md|
+  Confirms migration of a contract to the sender address.
+  After a successful call the contract will be set to migrated state where no operations are possible.
+  |]
+
+proposeDoc :: Markdown
+proposeDoc = [md|
+  Saves the proposal with specific id and freezes the ammount of sender tokens equal to the given ammount.
+  Sender must have enough unfrozen tokens.
+  The sender's ammount of frozen tokens is increased by proposal ammount. And the amount of unfrozen one
+  is decreased by the same value.
+  |]
+
+voteDoc :: Markdown
+voteDoc = [md|
+  For each vote params in a given list votes for the proposal with that id. Thus the sender can vote many proposal
+  (or one proposal multiple times) in a single call.
+  Sender must have an ammount required for all votings.
+  |]
+
+setVotingPeriodDoc :: Markdown
+setVotingPeriodDoc = [md|
+  Updates how long the voting period would last.
+  Can be performed only by administrator
+  It affects all ongoing proposals and all created afterwards.
+  |]
+
+setQuorumThresholdDoc :: Markdown
+setQuorumThresholdDoc = [md|
+  Updates the quorum threshold with a given value.
+  Can be performed only by administrator.
+  It affects all ongoing proposals and all created afterwards.
+  |]
+
+
+flushDoc :: Markdown
+flushDoc = [md|
+  Finish voting process on all proposals where the voting period is over.
+  Returns an ammount to the proposer, determined by the result of voting.
+  There is a possibility of some tokens being lost due to administrator perform
+  of burn or transfer operation.
+  If proposal is accepted, the decision lambda is called.
+  |]
+
+burnDoc :: Markdown
+burnDoc = [md|
+  Reduces the amount of tokens of the given address. Can be performed only by administrator
+  and only if the given address have enought tokens to burn.
+  |]
+
+mintDoc :: Markdown
+mintDoc = [md|
+  Provides the amount of tokens of the given address.
+  Can be performed only by administrator.
+  |]
+
+transferContractTokensDoc :: Markdown
+transferContractTokensDoc = [md|
+  This entrypoint can be used by the administrator
+  to transfer tokens owned (or operated) by this contract in another FA2 contract.
+  Can be performed only by administrator.
+  |]
+
+tokenAddressDoc :: Markdown
+tokenAddressDoc = [md|
+  Returns the address of the associated FA2 contract.
+  Since FA2 logic is embedded into this contract, this entrypoint always returns SELF.
+  |]
```

Comment:
In one of the projects, we had a mandatory rule, that if someone writes documentation he has to spellcheck it.

So could you pass the entire text into some spellchecker and ensure that checks are passed? There is https://grammarly.com, and even grammarly plugin to VSCode, or Idea (though not sure whether does it check grammar) - whatever seems more convenient.

Reply:
>  Idea has build-in spellchecker 

It doesn't check Markdown for some reason

Reply:
Oh mda, good grief indeed